### PR TITLE
Replaced .gitignore file with an improved version

### PR DESCRIPTION
**Pull request by gamerover98**
This version of .gitignore allows you to add what do you want to but not the IDE files and compiled stuff

### CODE REVIEW

**Comment by gamerover98** [comment]
### Improved CommandHandler class code to guarantee OOP and added an addCommand method to add command with a custom CommandHandler instance

You can now set a different CommandHandler instance for your command implementation.
Anyway, I don't know how to edit: `@Contract("_ -> param1")` located on the top of the registerPluginCommand method. Let me know!

**Review by TozyMC — COMMENTED** [review]
The `.gitignore` file is generated automatically by IntellJ IDEA. It will ignore the build files and personal files. Never mind!!!

File: src/main/java/xyz/tozymc/spigot/api/command/CommandController.java [hunk 1]
```diff
@@ -69,15 +69,15 @@ public Command getRootCommand(@NotNull String name) {
   }
 
   @NotNull
-  @Contract("_ -> param1")
-  private Command registerPluginCommand(@NotNull Command command) {
+  @Contract("_ -> param1") // <-- TODO: I don't know how to edit this Contract attribute
```
[the remark this answers is not in the document]
Comment (gamerover98) [reply src/main/java/xyz/tozymc/spigot/api/command/CommandController.java:72]: How?

File: src/main/java/xyz/tozymc/spigot/api/command/CommandController.java [hunk 2]
```diff
@@ -69,15 +69,15 @@ public Command getRootCommand(@NotNull String name) {
...
+  private Command registerPluginCommand(@NotNull Command command, @NotNull CommandHandler commandHandler) {
```
[the remark this answers is not in the document]
Comment (gamerover98) [reply src/main/java/xyz/tozymc/spigot/api/command/CommandController.java:73]: How do you want to solve this?

**Review by TozyMC — COMMENTED** [review]
Commit 75c0498bcac5c07cc9a31876c6f73965bad33eb1 was resolved [this question](https://github.com/TozyMC/commands-api/pull/2#discussion_r519404542)

The style guide based on Java Google Style Guide. It was edited package grouping.

**Pull request merged**